x-pack/plugins/security/server/authentication/authenticator.ts

```diff
@@ -455,6 +478,32 @@ export class Authenticator {
     return [...this.providers.values()].some(provider => provider.type === providerType);
   }
 
+  /**
+   * Acknowledges access agreement on behalf of the currently authenticated user.
+   * @param request Request instance.
+   */
+  async acknowledgeAccessAgreement(request: KibanaRequest) {
+    assertRequest(request);
+
+    const sessionStorage = this.options.sessionStorageFactory.asScoped(request);
+    const existingSession = await this.getSessionValue(sessionStorage);
+    const currentUser = this.options.getCurrentUser(request);
+    if (!existingSession || !currentUser) {
+      throw new Error('Cannot acknowledge access agreement for unauthenticated user.');
+    }
+
+    if (!this.options.license.getFeatures().allowAccessAgreement) {
+      throw new Error('Current license does not allow access agreement acknowledgement.');
```

Comment:
__note:__ this doesn't sound like an audit event, but rather as a developer mistake, hence just throwing error here.